examples/mplot3d/mixed_subplots_demo.py

```diff
@@ -12,38 +13,42 @@ def f(t):
     return np.multiply(s1, e1)
 
 
-################
+#############################################
+# Set up a figure twice as tall as it is wide
+#############################################
+fig = plt.figure(figsize=plt.figaspect(2.))
+fig.suptitle('A tale of 2 subplots')
+
+
+#############################################
 # First subplot
-################
+#############################################
+ax = fig.add_subplot(2, 1, 1)
+
 t1 = np.arange(0.0, 5.0, 0.1)
 t2 = np.arange(0.0, 5.0, 0.02)
 t3 = np.arange(0.0, 2.0, 0.01)
 
-# Twice as tall as it is wide.
-fig = plt.figure(figsize=plt.figaspect(2.))
-fig.suptitle('A tale of 2 subplots')
-ax = fig.add_subplot(2, 1, 1)
-l = ax.plot(t1, f(t1), 'bo',
-            t2, f(t2), 'k--', markerfacecolor='green')
+ax.plot(t1, f(t1), 'bo',
+        t2, f(t2), 'k--', markerfacecolor='green')
 ax.grid(True)
 ax.set_ylabel('Damped oscillation')
 
 
-#################
+#############################################
 # Second subplot
-#################
+#############################################
 ax = fig.add_subplot(2, 1, 2, projection='3d')
+
 X = np.arange(-5, 5, 0.25)
-xlen = len(X)
 Y = np.arange(-5, 5, 0.25)
-ylen = len(Y)
 X, Y = np.meshgrid(X, Y)
 R = np.sqrt(X**2 + Y**2)
 Z = np.sin(R)
 
 surf = ax.plot_surface(X, Y, Z, rstride=1, cstride=1,
                        linewidth=0, antialiased=False)
-
 ax.set_zlim3d(-1, 1)
```

Comment:
could we update this to be `set zlim(-1, 1)` instead? The "3d" part is an anachronism.